### DISCUSSION TITLE
Please vote: API changes in LinkAce 2

### DISCUSSION THREAD

**Discussion by Kovah** [Kovah/LinkAce#556] [Polls]
Some of you might follow the discussions and updates about support for multiple users in LinkAce v2 (#165). One part of this feature is API access for **all** data that is stored in LinkAce via system API access. API tokens are needed to authorize yourself when accessing the API.

#### Status quo
At the moment you can generate an API token for yourself and access all data in LinkAce, because you are the owner of everything: links, lists, tags, notes,...

#### The problem
With support for multiple users comes an advanced permission system. Users can create private links, lists and so on. Obviously, this should not be visible to _any_ other user. When accessing the API, users will most likely not be able to access all data on LinkAce.

#### Idea for LinkAce 2
Users have their private API tokens and can access all data that is visible to them. Private links from other users are not accessible via the API.
However, administrators can generate **system API tokens**, which can be granted access to **all** data, including private links, lists,... of all users.

**Comment by clem3109** [comment]
Personally, I use an API token in an iOS shortcut, but if there is any chance that someone is working on a safari extension, I would have no use of it.

**Reply by Kovah** [reply]
But would you need access to all user data in LinkAce, including private links and lists? I would think that a browser extension only accesses data you can also view in your browser. I updated the poll to clarify this.

**Reply by clem3109** [reply]
Right, I just add links from it and see the tags and the lists. 
I just copied and improved a bit a shortcut from one of your blog post

**Comment by Kovah** [comment]
Poll is closed now. LinkAce v2 will not ship with support for system-wide API access.
